### ISSUE TITLE
Better use of cffi (>=1.0)

### ISSUE THREAD

**Issue by dimaqq** [xattr/xattr#41]
Hey I see that ffi.verify is called at top level in lib which is imported directly into init. If I understand it right, this precludes a project that uses xattr to be wrapped in pyinstaller or similar.

Please check these two posts, apparently CFFI v1.0 improved things a log, and perhaps a few simple changes can get this package in line?

https://caremad.io/2014/11/distributing-a-cffi-project/
https://caremad.io/2015/06/distributing-a-cffi-project-redux/

**Comment by etrepum** [comment]
Have you had a problem using xattr in practice? This is the first mention of this potential issue that I've seen. /cc @alex

**Comment by alex** [comment]
The biggest practical benefit here would be faster import times.

**Comment by devinlane** [comment]
Faster import times would be quite nice, as on my system xattr takes nearly 100ms to import, far greater than all of the other libraries I'm using combined.

I dug around and was able to figure out the necessary changes:

1. Put the source/cdef in lib_build.py
2. Use ffi.set_source() instead of ffi.verify():
```python
C_SRC = '''<all the source>'''
ffi.set_source('_lib', C_SRC)
```
3. Run lib_build.py, presumably as part of setup. Note that I have 0 experience using setuptools or distributing packages, so not quite sure how this needs to be done.
4. In lib.py, import as follows:

```python
try:
    from ._lib import lib
except ImportError:
    from .lib_build import ffi, C_SRC
    lib = ffi.verify(C_SRC)
```

This will fallback to using verify if for some reason the native module wasn't built.

Using this method I can import xattr in ~1ms.

Would be happy to help get this in as part of xattr if someone can help with the setuptools part :)

**Comment by etrepum** [comment]
Fixed by #47
